common/models/zod-schemas.ts

```diff
@@ -55,3 +56,44 @@ export const OttApiRequestAccountRecoveryVerifySchema = z.object({
 	verifyKey: z.string(),
 	newPassword: z.string(),
 });
+
+const CategorySchema = z.enum([
+	"sponsor",
+	"intro",
+	"outro",
+	"interaction",
+	"selfpromo",
+	"music_offtopic",
+	"preview",
+]);
+
+const RoomSettingsSchema = z.object({
+	name: z.string().optional(),
+	title: z.string().optional(),
+	description: z.string().optional(),
+	visibility: z.nativeEnum(Visibility),
+	queueMode: z.nativeEnum(QueueMode),
+	autoSkipSegmentCategories: z.array(CategorySchema),
+	restoreQueueBehavior: z.nativeEnum(BehaviorOption),
+	enableVoteSkip: z.boolean().optional(),
+});
+
+export const OttApiRequestPatchRoomSchema = z.union([
+	z.object({
+		grants: z.array(
+			z.tuple([
+				z.object({
+					role: z.nativeEnum(Role),
+				}),
+				z.object({
+					grantMask: z.number(),
+				}),
+			])
+		),
+		...RoomSettingsSchema.shape,
+	}),
+	z.object({
+		claim: z.boolean().optional(),
+		...RoomSettingsSchema.shape,
```

Comment:
and `RoomSettingsSchema` should be excluded from here.

Reply:
To clarify, did you mean they should be excluded from both? Or is how it is now fine?

Still needs to be a union type right? So union type between the claim and the rest of the room settings fields?